pandas/tslib.pyx

```diff
@@ -1220,7 +1220,10 @@ cdef class _Timestamp(datetime):
             return Timestamp((self.freq * other).apply(self), freq=self.freq)
 
         elif isinstance(other, timedelta) or hasattr(other, 'delta'):
-            nanos = _delta_to_nanoseconds(other)
+            try:
```

Comment:
Cython really doesn't let you raise any errors from `__add__` without calling `__radd__` on the other object? That is surprising -- and disappointing!

It *is* true that Cython will use `__add__` for both Python's `__add__` and `__radd__`, so `self` in this method may not be a `Timestamp` instance:
http://cython.readthedocs.io/en/latest/src/userguide/special_methods.html#arithmetic-methods